apple/Sources/FerrostarCore/Mock/TestFixtureFactory.swift

```diff
@@ -0,0 +1,92 @@
+import FerrostarCoreFFI
+import Foundation
+
+public protocol TestFixtureFactory {
```

Comment:
I might have gone overboard here, but I expect it to pay dividends to make it easier to create scaffolding data for our previews and unit tests. If you think it's too much, I can inline something less clever for my specific use case.

Reply:
Oh boy 😂 I have an allergic reaction to seeing the term "factory" in a Swift codebase, but I'll try to come up with some actually constructive feedback after I've had some time to look at the rest of the changes hahaha.

Reply:
Would a factory by any other name smell as bad?

Reply:
I'm mostly joking; PTSD from working in the JVM ecosystem haha. This describes what it does, and it's test code, not the public API, so I'm fine with it haha.

Reply:
Edit: Can this be internal? For tests, we can use testable imports.

Reply:
Figured out a resolution for this. It's used only for previews, so we can just move this to the `FerrostarSwiftUI` module and make everything in the file internal :)